research/cifar10/ditto/server.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+from functools import partial
+from logging import INFO
+from typing import Any, Dict
+
+import flwr as fl
+from flwr.common.logger import log
+from flwr.common.typing import Config
+from flwr.server.client_manager import SimpleClientManager
+from flwr.server.strategy import FedAvg
+
+from fl4health.utils.config import load_config
+from fl4health.utils.metric_aggregation import evaluate_metrics_aggregation_fn, fit_metrics_aggregation_fn
+from fl4health.utils.parameter_extraction import get_all_model_parameters
+from fl4health.utils.random import set_all_random_seeds
+from research.cifar10.model import ConvNet
+from research.cifar10.personal_server import PersonalServer
+
+
+def fit_config(
+    batch_size: int,
+    local_epochs: int,
+    n_server_rounds: int,
+    n_clients: int,
+    current_server_round: int,
+) -> Config:
+    return {
+        "batch_size": batch_size,
+        "local_epochs": local_epochs,
+        "n_server_rounds": n_server_rounds,
+        "n_clients": n_clients,
+        "current_server_round": current_server_round,
+    }
+
+
+def main(config: Dict[str, Any], server_address: str) -> None:
+    # This function will be used to produce a config that is sent to each client to initialize their own environment
+    fit_config_fn = partial(
+        fit_config,
+        config["batch_size"],
+        config["local_epochs"],
+        config["n_server_rounds"],
+        config["n_clients"],
+    )
+
+    client_manager = SimpleClientManager()
+    # Initializing the model on the server side
+    model = ConvNet(in_channels=3)
```

Comment:
Just to confirm, this is a similar or the same model that was used in the benchmark? Just trying to eliminate any areas for departure.

Reply:
I rechecked this, I actually got the model from their implementation and ConvNet2 was standing for 2 layer convent. Though one difference I found that in paper they said they were using hidden size of 512 for cifar10 dataset, and I updated model based on that.